src/mpid/ch4/shm/xpmem/xpmem_control.h

```diff
@@ -142,6 +163,166 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_XPMEM_ctrl_send_lmt_req_cb(MPIDI_SHM_ctrl_hdr
         XPMEM_PT2PT_DBG_PRINT("send_lmt_req_cb: enqueue unexpected, rreq=%p\n", rreq);
     }
 
+  fn_exit:
+    MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_XPMEM_CTRL_DO_SEND_LMT_REQ_CB);
+    return mpi_errno;
+  fn_fail:
+    goto fn_exit;
+}
+
+
+#undef FUNCNAME
+#define FUNCNAME MPIDI_XPMEM_ctrl_send_lmt_ack_cb
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+MPL_STATIC_INLINE_PREFIX int MPIDI_XPMEM_ctrl_send_lmt_ack_cb(MPIDI_SHM_ctrl_hdr_t * ctrl_hdr)
+{
+    int mpi_errno = MPI_SUCCESS;
+    MPIR_Request *sreq = (MPIR_Request *) ctrl_hdr->xpmem_slmt_ack.sreq_ptr;
+
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_XPMEM_CTRL_SEND_LMT_ACK_CB);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_XPMEM_CTRL_SEND_LMT_ACK_CB);
+
+    XPMEM_PT2PT_DBG_PRINT("send_lmt_ack_cb: complete sreq %p\n", sreq);
+    MPID_Request_complete(sreq);
+
+  fn_exit:
+    MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_XPMEM_CTRL_SEND_LMT_ACK_CB);
+    return mpi_errno;
+  fn_fail:
+    goto fn_exit;
+}
+
+#ifdef MPIDI_CH4_SHM_XPMEM_COOP_P2P
+#undef FUNCNAME
+#define FUNCNAME MPIDI_XPMEM_ctrl_recv_lmt_ack_cb
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+MPL_STATIC_INLINE_PREFIX int MPIDI_XPMEM_ctrl_recv_lmt_ack_cb(MPIDI_SHM_ctrl_hdr_t * ctrl_hdr)
+{
+    int mpi_errno = MPI_SUCCESS;
+    MPIR_Request *rreq = (MPIR_Request *) ctrl_hdr->xpmem_slmt_ack.sreq_ptr;
+    MPIDI_XPMEM_am_dmessage_t *dmessage;
+    MPIDI_SHM_ctrl_xpmem_send_lmt_req_t *slmt_req_hdr;
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_XPMEM_CTRL_RECV_LMT_ACK_CB);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_XPMEM_CTRL_RECV_LMT_ACK_CB);
+
+    XPMEM_PT2PT_DBG_PRINT("recv_lmt_ack_cb: complete rreq %p\n", rreq);
+
+    if (MPIDI_XPMEM_REQUEST(rreq, counter)) {
+        MPIR_Datatype_release_if_not_builtin(MPIDIG_REQUEST(rreq, datatype));
+        OPA_store_int(MPIDI_XPMEM_REQUEST(rreq, counter), 0);
+    }
+
+    MPID_Request_complete(rreq);
+
+    if (MPIDI_XPMEM_global.dmessage_queue) {
+        dmessage = MPIDI_XPMEM_global.dmessage_queue;
+        slmt_req_hdr = &dmessage->ctrl_hdr.xpmem_slmt_req;
+        if (!OPA_load_int
```

Comment:
Putting the line breaks in this way is very confusing. It makes it unclear that `OPA_load_int` is a function. Can you combine the next line with this one?